assets/l10n/app_en.arb

```diff
@@ -27,18 +31,298 @@
   "@profileButtonSendDirectMessage": {
     "description": "Label for button in profile screen to navigate to DMs with the shown user."
   },
-  "cameraAccessDeniedTitle": "Permissions needed",
-  "@cameraAccessDeniedTitle": {
-    "description": "Title for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededError": "Error",
+  "@permissionsNeededError": {
+    "description": "General title for access dialog when we encounter an unknown error."
   },
-  "cameraAccessDeniedMessage": "To upload an image, please grant Zulip additional permissions in Settings.",
-  "@cameraAccessDeniedMessage": {
-    "description": "Message for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededTitle": "Permissions needed",
+  "@permissionsNeededTitle": {
+    "description": "Title for dialog when the user needs to grant additional permissions."
   },
-  "cameraAccessDeniedButtonText": "Open settings",
-  "@cameraAccessDeniedButtonText": {
+  "permissionsNeededOpenSettings": "Open settings",
+  "@permissionsNeededOpenSettings": {
+    "description": "Button label for permissions dialog button that opens the system settings screen."
+  },
+  "permissionsNeededCameraAccessDenied": "To upload an image, please grant Zulip additional permissions in Settings.",
+  "@permissionsNeededCameraAccessDenied": {
     "description": "Message for dialog when the user needs to grant permissions for camera access."
   },
+  "permissionsReadExternalStorageDenied": "To upload files, please grant Zulip additional permissions in Settings.",
+  "@permissionsReadExternalStorageDenied": {
```

Comment:
These should be consistent about whether "needed" is a part of their naming prefix.

Reply:
The "needed" does feel kind of redundant with "denied". Perhaps `permissionDeniedFooBar`?